agent/handlers/agentapi/v1/taskprotection/handlers/handlers.go

```diff
@@ -0,0 +1,136 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
+	"github.com/aws/amazon-ecs-agent/agent/handlers/agentapi/v1/taskprotection/types"
+	"github.com/aws/amazon-ecs-agent/agent/handlers/utils"
+	v3 "github.com/aws/amazon-ecs-agent/agent/handlers/v3"
+	"github.com/cihub/seelog"
+)
+
+// Returns endpoint path for PutTaskProtection API
+func TaskProtectionPath() string {
+	return fmt.Sprintf(
+		"/api/v1/%s/task/protection",
+		utils.ConstructMuxVar(v3.V3EndpointIDMuxName, utils.AnythingButSlashRegEx))
+}
+
+// Task protection request received from customers pending validation
+type taskProtectionRequest struct {
+	ProtectionType           string
+	ProtectionTimeoutMinutes *int
+}
+
+// PutTaskProtectionHandler returns an HTTP request handler function for
+// PutTaskProtection API
+func PutTaskProtectionHandler(state dockerstate.TaskEngineState,
+	cluster string) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		putTaskProtectionRequestType := "v1/PutTaskProtection"
```

Comment:
nit. maybe we can move constants to a constant block 
``` 
const (
	putTaskProtectionRequestType = "v1/PutTaskProtection"
        getTaskProtectionRequestType := "api/v1/GetTaskProtection"
)

Reply:
and do we need  to append `api/` for putTaskProtectionRequestType?

Reply:
Yeah I will make the two constants consistent.

Reply:
Hmm json decoding doesn't work if fields are not exported...

Reply:
I think I will export the type and use it to define requests in `task_server_setup_test.go`.